src/Type/Constant/ConstantArrayType.php

```diff
@@ -458,17 +458,41 @@ public function equals(Type $type): bool
 
 	public function isCallable(): TrinaryLogic
 	{
-		$typeAndMethods = $this->findTypeAndMethodNames();
-		if ($typeAndMethods === []) {
+		$callableArray = $this->getClassOrObjectAndMethods();
+		if ($callableArray === []) {
 			return TrinaryLogic::createNo();
 		}
 
-		$results = array_map(
-			static fn (ConstantArrayTypeAndMethod $typeAndMethod): TrinaryLogic => $typeAndMethod->getCertainty(),
-			$typeAndMethods,
-		);
+		[$classOrObject, $methods] = $callableArray;
+		if (count($methods->getConstantStrings()) === 0) {
+			return TrinaryLogic::createMaybe();
+		}
+
+		$type = $classOrObject->getObjectTypeOrClassStringObjectType();
+		if (!$type->isObject()->yes()) {
+			return TrinaryLogic::createMaybe();
+		}
+
+		$hasMethodTrinary = [];
+		$phpVersion = PhpVersionStaticAccessor::getInstance();
+		foreach ($methods->getConstantStrings() as $method) {
+			$has = $type->hasMethod($method->getValue());
+
+			if ($has->yes()) {
+				if (!$phpVersion->supportsCallableInstanceMethods()) {
+					$methodReflection = $type->getMethod($method->getValue(), new OutOfClassScope());
+					if ($classOrObject->isString()->yes() && !$methodReflection->isStatic()) {
+						$has = $has->and(TrinaryLogic::createNo());
```

Comment:
In `findTypeAndMethodNames` there is a `continue;` I don't want to.

Reply:
Why not `$has = TrinaryLogic::createNo()`?